From a03e3a7663a26debb0ddac82a69b5cd90eb9d836 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 24 Sep 2020 06:50:54 -0700
Subject: [PATCH 1658/1852] dmaengine: xilinx: frmbuf: Fix warnings for
 PW.MIXED_ENUM_TYPE

commit 4fe2ed5fdc4f2ed2aa8dd47615cd6066487c3940 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the PW.MIXED_ENUM_TYPE type.

The actual warnings are:

In xilinx_frmbuf_probe():
Event mixed_enum_type: enumerated type mixed with another type

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_frmbuf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/xilinx/xilinx_frmbuf.c b/drivers/dma/xilinx/xilinx_frmbuf.c
index b6766da4509e..5ab1a1e66132 100644
--- a/drivers/dma/xilinx/xilinx_frmbuf.c
+++ b/drivers/dma/xilinx/xilinx_frmbuf.c
@@ -1577,7 +1577,7 @@ static int xilinx_frmbuf_probe(struct platform_device *pdev)
 		align = 16;
 	}
 
-	xdev->common.copy_align = fls(align) - 1;
+	xdev->common.copy_align = (enum dmaengine_alignment)(fls(align) - 1);
 	xdev->common.dev = &pdev->dev;
 
 	if (xdev->cfg->flags & XILINX_CLK_PROP) {
-- 
2.31.1

